app.js

```diff
@@ -12,8 +13,34 @@ class Dashboard extends React.Component {
     render() {
         return (
             <div>
-                <PsnrResultsChart />
-                <FrameSkipResultsChart />
+                <Chart
+                    graphTitle='PSNR'
+                    graphYAxis='PSNR value'
+                    graphXAxis='Build number'
+                    config={{
+                        resultsUrl: './psnrResults',
+                        resultTransformers: [ psnrTransformer ]
+                    }}
+                />
+
+                <Chart
+                    graphTitle='Frozen/skipped frames'
+                    graphYAxis='% of total frames'
+                    graphXAxis='Build number'
+                    config={{
+                        resultsUrl: './psnrResults',
```

Comment:
This is an interesting tidbit. Two requests are made to the same endpoint?

Reply:
yeah, this felt a bit weird but not doing it this way means we'd have to do something like cache it and have a layer which would transparently check the cache.  if you've got a good idea let me know.  i don't think it's a dealbreaker for this, but it would be nice to be able to avoid it.

Reply:
Creating a model that does the request and parses out various results. The model doesn't have to be a conventional model in the sense of mvc. Dashboard (or whatever other view) calls fetch through some api on the model, which returns a promise, then you can set state to update the chart display.